src/test/java/catan/controllers/testcases/play/BuyDevCardTest.java

```diff
@@ -0,0 +1,360 @@
+        playPreparationStage()
+
+                .THROW_DICE(1)
+
+                .BUY_CARD(1).successfully()
+                .BUY_CARD(1).successfully();
+    }
+
+    @Test
+    public void should_return_bought_card_in_response() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+                .nextRandomDevelopmentCards(asList(KNIGHT, VICTORY_POINT, ROAD_BUILDING, MONOPOLY, YEAR_OF_PLENTY))
+
+                .BUY_CARD(1).boughtCardIs(KNIGHT)
+                .BUY_CARD(1).boughtCardIs(VICTORY_POINT)
+                .BUY_CARD(1).boughtCardIs(ROAD_BUILDING)
+                .BUY_CARD(1).boughtCardIs(MONOPOLY)
+                .BUY_CARD(1).boughtCardIs(YEAR_OF_PLENTY);
+    }
+
+    @Test
+    public void should_increase_development_cards_when_user_buys_a_new_one() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+                .nextRandomDevelopmentCards(asList(KNIGHT, KNIGHT, ROAD_BUILDING, KNIGHT, YEAR_OF_PLENTY, YEAR_OF_PLENTY))
+                .startTrackDevCardsQuantity()
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(1, 0, 0, 0, 0)
+                .getGameDetails(2).gameUser(2).devCardsQuantityChanged(0, 0, 0, 0, 0)  // no cards should be increased for other players
+                .getGameDetails(3).gameUser(3).devCardsQuantityChanged(0, 0, 0, 0, 0)  // no cards should be increased for other players
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(1, 0, 0, 0, 0)
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(0, 0, 1, 0, 0)
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(1, 0, 0, 0, 0)
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(0, 0, 0, 0, 1)
+
+                .BUY_CARD(1)
+                .getGameDetails(1).gameUser(1).devCardsQuantityChanged(0, 0, 0, 0, 1);
+    }
+
+    @Test
+    public void should_fail_when_cards_are_over() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1)
+                .BUY_CARD(1).successfully()
```

Comment:
i think it would be good if you check the quantity of user's development cards here. it would represent that quantity of available development cards was correct.